[1/N refactor NCCLPG error logs] Make NCCL PG error messages more accurate and simpler

fduwjj opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #134036
* __->__ #134017

We did a thorough review on all the error messages we are logging inside PGNCCL, and we want to make log message simpler and more accurate, this is the first PR for this effort.


cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @wz337 @wconstab @d4l3k @c-p-i-o

fduwjj committed:
[1/N] Make NCCL PG error messages more accurate and simpler

[ghstack-poisoned]

fduwjj commented on torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:
```diff
@@ -1742,8 +1740,9 @@ void ProcessGroupNCCL::watchdogHandler() {
-                         << "First watchdog to set the dump signal.";
+              LOG(ERROR)
+                  << logPrefix()
+                  << "First watchdog to set the dump signal to TCPStore.";
```
I think it's other ranks, but I agree with the broadcasting part.

fduwjj committed:
Update on "[1/N] Make NCCL PG error messages more accurate and simpler"


We did a thorough review on all the error messages we are logging inside PGNCCL, and we want to make log message simpler and more accurate, this is the first PR for this effort.


cc XilunWu H-Huang awgu kwen2501 wanchaol fegin wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]

wconstab commented on torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:
```diff
@@ -1732,8 +1729,9 @@ void ProcessGroupNCCL::watchdogHandler() {
+        // try to notify other ranks via global TCPStore to dump the flight
+        // recorder if exception happens. Flight recorder behavior is
+        // independent of desync Debug.
         if (dumpOnException_) {
```
nit: some folks got confused by 'exception' here, iirc.  it is actually timeout or exception, isn't it?

fduwjj replied:
Sure, I also changed the variable name, lol

wconstab commented on torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:
```diff
@@ -1742,8 +1740,9 @@ void ProcessGroupNCCL::watchdogHandler() {
-                         << "First watchdog to set the dump signal.";
+              LOG(ERROR)
+                  << logPrefix()
+                  << "Broadcasting the dump signal of flight-recorder to other processes via TCPStore.";
```
nit: "flight-recorder dump signal" seems clearer than "dump signal of flight-recorder"

fduwjj committed:
Update on "[1/N] Make NCCL PG error messages more accurate and simpler"


We did a thorough review on all the error messages we are logging inside PGNCCL, and we want to make log message simpler and more accurate, this is the first PR for this effort.


cc XilunWu H-Huang awgu kwen2501 wanchaol fegin wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]

fduwjj commented:
@pytorchbot merge

fduwjj commented:
@pytorchbot merge -i
